dev/tools/merge-pr.sh

```diff
@@ -193,9 +195,26 @@ if [ "$KIND" = "[]" ]; then
 fi
 
 # Sanity check: user.signingkey
-if [ -z "$(git config user.signingkey)" ]; then
-  warning "git config user.signingkey is empty"
-  warning "gpg will guess a key out of your git config user.* data"
+GIT_SIGNINGKEY="$(git config user.signingkey)"
+if [ -z "$GIT_SIGNINGKEY" ]; then
+  error "git config user.signingkey is empty"
+  error "gpg will guess a key out of your git config user.* data"
+  ask_confirmation
+fi
+
+# Sanity check: user.signingkey exists in gpg and matches user.email
+# Get data from GIT configuration
+GIT_EMAIL="$(git config user.email | tr '[:upper:]' '[:lower:]')"
+# Ask GPG for the email for this key
+# See usr\share\doc\gnupg2\DETAILS for documentation of the --with-colons format
+# The returned ID is assumed to be of the form "name <email>", from which the email is extracted
+# In case the ID ist just an email, the sed is a NOP
+GPG_SIGNINGKEY_TYPE="$(gpg --list-secret-keys --with-colons "$GIT_SIGNINGKEY" | cut --delimiter=: --fields=1)"
+GPG_SIGNINGKEY_EMAIL="$(gpg --list-secret-keys --with-colons "$GIT_SIGNINGKEY" | cut --delimiter=: --fields=10 | sed -e 's|.*<\(.*\)>|\1|' | tr '[:upper:]' '[:lower:]')"
+if [ "$GPG_SIGNINGKEY_TYPE" != "sec" ] || [ "$GPG_SIGNINGKEY_EMAIL" != "$GIT_EMAIL" ]; then
```

Comment:
Does this check work if the key has several e-mail addresses attached?

Reply:
In fact, maybe I can test that.

Reply:
I didn't know that this is possible, but some googling revealed how to add a second email to an existing private key. It will likely not work. I will fix this.

Reply:
Indeed, I have such a setup and the check fails on my machine.

Reply:
Would you mind sending me the output of the below command via private mail:

```
gpg --list-secret-keys --with-colons <yourkeyid> | cut --delimiter=: --fields=10
```

Reply:
done!